payments/lightning-payments.md

```diff
@@ -98,12 +98,35 @@ _(All other examples below, trust not to reveal secret preimage to any other rou
 
     _**Caveat:** if a path participant sits on two routes, there's a risk they can take the revealed preimage from payment of one invoice and collect settlement on the second invoice before the recipient is able to settle their hodl invoice_
 
+#### Limitations
+...
+
 
 ### Keysend (no invoice)
 
-The `keysend` feature is a way for a sender to send a payment to a given recipient over the Lightning Network **without first having an invoice** from the recipient to send against.
+The `keysend` feature is a way for a sender to send a payment to a given recipient over the Lightning Network **without first having an invoice** from the recipient to send against ([docs](https://lightning.readthedocs.io/lightning-keysend.7.html)).
+
+The way this works is the sender first needs the **destination node id (node's public key)** for the recipient's Lightning Network node. The sender then creates a preimage and uses it to construct a keysend payment that it then forwards along an appropriate route to the recipient.
+
+When the recipient receives the keysend payment, they would decode the payment and extract the preimage to accept the payment (resolve the HTLC). The recipient would also create an ad-hoc invoice on their end that corresponds to the payment and store it in their local invoice registry.
+
+The recipient would usually have no way of knowing who sent the keysend payment unless the sender somehow makes this explicitly known.
```

Comment:
what mechanisms exist to make the sender "explicitly known"?

Reply:
[Just added some links](https://github.com/peakshift/bitcoin-ux/blob/lightning-payments/payments/lightning-payments.md#useful-additional-data-mechanisms--applications) (https://github.com/peakshift/bitcoin-ux/pull/28/commits/32326ed7ea312b92726b8a97961e600c0e4e5063) that would relate to this. Pretty much it would be to make use of "custom data records" somehow. A good example is how `whatsat` actually encodes this (linked in the section)